### PR TITLE
Add vscode settings. format files

### PR DESCRIPTION
**Pull request by Woz4tetra**
Add format on save setting to keep code looking consistent.

### CODE REVIEW

File: .vscode/extensions.json
Comment (nix342) [inline .vscode/extensions.json:1]: There's a lot here. Any particularly good ones on the list?

Comment (Woz4tetra) [reply .vscode/extensions.json:1]: I've pruned the list down. I was mostly copying from my local settings. The list now contains the extensions that are actually applicable to our codebase.

Comment (nix342) [reply .vscode/extensions.json:1]: Cool!

**Review by nix342 — APPROVED** [review]
Good stuff...formatting always drives me crazy at some point in the season. This will be nice.

**Pull request merged**